crates/rattler_networking/tests/integration_test.rs

```diff
@@ -0,0 +1,311 @@
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+use rattler_networking::{
+    s3_middleware::S3Config, AuthenticationMiddleware, AuthenticationStorage, S3Middleware,
+};
+use reqwest::Client;
+use rstest::*;
+use serial_test::serial;
+use tempfile::{tempdir, TempDir};
+use url::Url;
+
+/* -------------------------------------- UTILS ------------------------------------- */
+
+struct MinioServer {
+    handle: std::process::Child,
+    #[allow(dead_code)]
+    directory: tempfile::TempDir,
+}
+
+impl MinioServer {
+    fn new() -> Self {
+        let directory = tempfile::tempdir().expect("Failed to create temp directory");
+        let args = [
+            "server",
+            directory.path().to_str().unwrap(),
+            "--address",
+            "127.0.0.1:9000",
+        ];
+        let handle = std::process::Command::new("minio")
+            .args(args)
+            .spawn()
+            .expect("Failed to start Minio server");
+        eprintln!(
+            "Starting Minio server with args (PID={}): {:?}",
+            handle.id(),
+            args
+        );
+        MinioServer { handle, directory }
+    }
+}
+
+impl Drop for MinioServer {
+    fn drop(&mut self) {
+        eprintln!("Shutting down Minio server (PID={})", self.handle.id());
+        match self.handle.kill() {
```

Comment:
it looks like this isn't working properly on windows :/

Reply:
Hm, then I'm going to rewrite the integration tests so that we no longer start minio as a subprocess from rust, but start it as a background service before running the integration tests.